packages/core/src/test/amazonqFeatureDev/prepareRepoData.test.ts

```diff
@@ -0,0 +1,114 @@
+import { prepareRepoData, TelemetryHelper } from '../../amazonqFeatureDev'
+import { AmazonqCreateUpload, getRandomString, Metric } from '../../shared'
+
+type resultType = {
+    zipFileBuffer: Buffer
+    zipFileChecksum: string
+}
+
+describe('prepareRepoDataPerformanceTest', function () {
+    function verifyResult(result: resultType, telemetry: TelemetryHelper, expectedSize: number): void {
+        assert.ok(result)
+        assert.strictEqual(Buffer.isBuffer(result.zipFileBuffer), true)
+        assert.strictEqual(telemetry.repositorySize, expectedSize)
+        assert.strictEqual(result.zipFileChecksum.length, 44)
+    }
+
+    performanceTest(
+        {
+            testRuns: 10,
+            linux: {
+                userCpuUsage: 80,
+                heapTotal: 4,
+                duration: 1.5,
+            },
+            darwin: {
+                userCpuUsage: 80,
+                systemCpuUsage: 35,
+                heapTotal: 4,
+                duration: 10,
+            },
+            win32: {
+                userCpuUsage: 80,
+                systemCpuUsage: 35,
+                heapTotal: 4,
+                duration: 3,
+            },
+        },
+        'handles many files',
+        function () {
+            const telemetry = new TelemetryHelper()
+            let workspace: WorkspaceFolder
+            let result: resultType
+            return {
+                setup: async () => {
+                    workspace = await createTestWorkspace(500, {
+                        fileNamePrefix: 'file',
+                        fileContent: '0123456789',
+                        fileNameSuffix: '.md',
+                    })
+                },
+                execute: async () => {
+                    result = await prepareRepoData([workspace.uri.fsPath], [workspace], telemetry, {
+                        record: () => {},
+                    } as unknown as Metric<AmazonqCreateUpload>)
+                },
+                verify: async () => verifyResult(result, telemetry, 5000),
+            }
+        }
+    )
+
+    performanceTest(
+        {
+            testRuns: 10,
+            linux: {
+                userCpuUsage: 65,
+                systemCpuUsage: 30,
+                heapTotal: 1,
+                duration: 0.2,
+            },
+            darwin: {
+                userCpuUsage: 50,
+                systemCpuUsage: 25,
+                heapTotal: 1,
+                duration: 0.4,
+            },
+            win32: {
+                userCpuUsage: 60,
+                systemCpuUsage: 30,
+                heapTotal: 1,
+                duration: 0.2,
+            },
+        },
+
+        'handles large files',
+        function () {
+            const telemetry = new TelemetryHelper()
+            let result: resultType
+            let workspace: WorkspaceFolder
+            return {
+                setup: async () => {
+                    workspace = await createTestWorkspace(10, {
```

Comment:
You can also return workspace from here and then use then use workspace as an arg to execute instead of having to declare things at the top level

Reply:
Yes, I like that better! I wasn't sure if this was only for stubs/spys and the like. After looking deeper at the implementation I realized I could do the same for result too.